access-common/src/main/java/edu/unc/lib/dl/ui/service/LorisContentService.java

```diff
@@ -167,23 +176,140 @@ public void streamJP2(String simplepid, String region, String size, String rotat
+        manifest.addMetadata("", "<a href=\"" +
+                URIUtil.join(basePath, "record", rootObj.getId()) + "\">View full record</a>");
+        String attribution = "University of North Carolina Libraries, Digital Collections Repository";
+        String collection = rootObj.getParentCollectionName();
+        if (collection != null) {
+            attribution += " - Part of " + collection;
+        }
+        manifest.addMetadata("Attribution", attribution);
+
+        Sequence seq = createSequence(manifestBase, briefObjs);
+
+        return iiifMapper.writeValueAsString(manifest.addSequence(seq));
+    }
+
+    public String getSequence(HttpServletRequest request, List<BriefObjectMetadata> briefObjs)
+            throws JsonProcessingException {
+        String path = getRecordPath(request);
+        return iiifMapper.writeValueAsString(createSequence(path, briefObjs));
+    }
+
+    public String getCanvas(HttpServletRequest request, BriefObjectMetadata briefObj)
+            throws JsonProcessingException {
+        String path = getRecordPath(request);
+        return iiifMapper.writeValueAsString(createCanvas(path, briefObj));
+    }
+
+    private Sequence createSequence(String seqPath, List<BriefObjectMetadata> briefObjs) {
+        Sequence seq = new Sequence(URIUtil.join(seqPath, "sequence", "normal"));
+
+        Set<String> uuidList = new HashSet<>();
+        for (BriefObjectMetadata briefObj : briefObjs) {
+            String datastreamUuid = jp2Pid(briefObj);
+
+            if (!datastreamUuid.equals("")) {
+                // Don't add rootObj twice
+                if (uuidList.contains(datastreamUuid)) {
+                    continue;
+                }
+
+                Canvas canvas = createCanvas(seqPath, briefObj);
+                seq.addCanvas(canvas);
+            }
+
+            uuidList.add(datastreamUuid);
+        }
+
+        return seq;
+    }
+
+    private Canvas createCanvas(String path, BriefObjectMetadata briefObj) {
+        String title = getTitle(briefObj);
+        String uuid = jp2Pid(briefObj);
+        Canvas canvas = new Canvas(path, title);
+        String canvasPath = URIUtil.join(basePath, "jp2Proxy", uuid, "jp2");
+
+        Datastream fileDs = briefObj.getDatastreamObject(DatastreamType.ORIGINAL_FILE.getId());
+        String extent = fileDs.getExtent();
+        if (extent != null) {
+            String[] imgDimensions = extent.split("x");
+            canvas.setHeight(Integer.parseInt(imgDimensions[0]));
+            canvas.setWidth(Integer.parseInt(imgDimensions[1]));
+        }
+
+        canvas.addIIIFImage(canvasPath, ImageApiProfile.LEVEL_TWO);
+        ImageContent thumb = new ImageContent(URIUtil.join(basePath,
+                "services", "api", "thumb", uuid, "large"));
+        canvas.addImage(thumb);
+
+        String label = briefObj.getLabel();
+        if (label != null) {
+            canvas.addLabel(label);
+        }
```

Comment:
I think this can be removed. `new Canvas(path, title)` adds the title as a label, and it looks like the label field isn't being populated in bxc5.